### ISSUE TITLE
Scripting

### ISSUE THREAD

**Issue by DrOptix** [DrOptix/ghost-engine#80]
Create a `ScriptComponent` that can be attached to an entity and it will run Rust code.

**Comment by DrOptix** [comment]
This one turns to be more complicated than expected. I'm sure the problem I face, dead locks or already mutable borrowed panics, are because of architectural decisions.

At the moment I can't figure out how to get out of this mess. I will postpone this to a later date and I will focus on other parts of the engine.

**Comment by DrOptix** [comment]
Unblocked as we have a prototype for executing the scripts. Details in this commit: 83e139d
